cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/TemplatesJarUtil.java

```diff
@@ -95,6 +97,43 @@ public static String downloadLatestDevon4jTemplates(boolean isDownloadSource, Fi
         TemplatesJarConstants.DEVON4J_TEMPLATES_ARTIFACTID, "LATEST", isDownloadSource, templatesDirectory);
   }
 
+  /**
+   * @param isDownloadSource true if downloading source jar file
+   * @param templatesDirectory directory where the templates jar are located
+   * @return fileName Name of the file downloaded
+   */
+  public static void downloadTemplates(boolean isDownloadSource, File templatesDirectory,
+      List<String> mavenCoordinates) {
+
+    if (mavenCoordinates != null && !mavenCoordinates.isEmpty()) {
+      for (String mcoordinate : mavenCoordinates) {
+        // simple check for maven naming conventions
+        mcoordinate = mcoordinate.trim();
+        if (!mcoordinate.matches("([a-zA-Z0-9_\\-\\.]+):([a-zA-Z0-9_-]+)(:([0-9]+(\\.[0-9]+)*(-SNAPSHOT)?))?")) {
+          throw new InvalidConfigurationException("configuration key:" + mcoordinate + " in .cobigen for "
+              + "template-sets.installed doesnt match the specification and could not be used");
+        }
+        String[] splited = mcoordinate.split(":");
+        String groupID = splited[0];
+        String artifactID = splited[1];
+        String version = splited.length > 2 ? splited[2] : null;
+        for (File fileOrDirectory : templatesDirectory.listFiles()) {
+          if (fileOrDirectory.isDirectory()) {
+            LOG.warn("Templates Folder already exists");
+            return;
+          } else if (fileOrDirectory.getName().contains(artifactID)) {
+            LOG.info("Template specified in the properties with the ArtifactID: " + artifactID + " already exits");
+            // Should we check for a update?
```

Comment:
@jan-vcapgemini @maybeec Should we check if an update for the template-sets-installed Templates is available automatically?